lstchain/irf/hdu_table.py

```diff
@@ -0,0 +1,345 @@
+    hdu_header["TELESCOP"] = "CTA"
+    hdu_header["INSTRUME"] = "LST-1"
+
+    filename_hdu_table = fits_dir/'hdu-index.fits.gz'
+
+    hdu = fits.BinTableHDU(hdu_table, header=hdu_header, name='HDU INDEX')
+    hdu_list = fits.HDUList([fits.PrimaryHDU(), hdu])
+    hdu_list.writeto(filename_hdu_table, overwrite=True)
+
+    obs_table = vstack(obs_tables)
+
+    obs_header = hdu_header.copy()
+    obs_header["HDUCLAS2"] = "OBS"
+    obs_header['MJDREFI'] = event_table.meta['MJDREFI']
+    obs_header['MJDREFF'] = event_table.meta['MJDREFF']
+
+    filename_obs_table = fits_dir/'obs-index.fits.gz'
+
+    obs = fits.BinTableHDU(obs_table, header = obs_header, name='OBS INDEX')
+    hdu_list = fits.HDUList([fits.PrimaryHDU(), obs])
+    hdu_list.writeto(filename_obs_table, overwrite=True)
+
+    return
+
+def create_event_list(data, run_number, source_name):
+    """
+    Create the event_list BinTableHDUs from the given data
+
+    Parameters
+    ----------
+        Data: DL2 data file
+                'astropy.table.QTable'
+        Run: Run number
+                Int
+        Source_name: Name of the source
+                Str
+    Returns
+    -------
+        Events HDU:  `astropy.io.fits.BinTableHDU`
+        GTI HDU:  `astropy.io.fits.BinTableHDU`
+        Pointing HDU:  `astropy.io.fits.BinTableHDU`
+    """
+
+    # Timing parameters
+    lam = 2800 # Average rate of triggered events, taken by hand for now
+    t_start = data['dragon_time'].value[0]
+    t_stop = data['dragon_time'].value[-1]
+    time = Time(data['dragon_time'], format='unix', scale="utc")
+    date_obs = time[0].to_value('iso', 'date')
+    obs_time = t_stop-t_start # All corrections excluded
+
+    # Position parameters
+    location = EarthLocation.from_geodetic(-17.89139 * u.deg, 28.76139 * u.deg, 2184 * u.m)
+    reco_alt = data['reco_alt']
+    reco_az = data['reco_az']
+    pointing_alt = data['pointing_alt']
+    pointing_az = data['pointing_az']
+
+    src_sky_pos = SkyCoord(alt=reco_alt, az=reco_az, frame=AltAz(obstime=time,
+                location=location)).transform_to(frame='icrs')
+    tel_pnt_sky_pos = SkyCoord(alt=pointing_alt[0], az=pointing_az[0], frame=AltAz(
+                obstime=time[0], location=location)).transform_to(frame='icrs')
+
+    try:
+        object_radec=SkyCoord.from_name(source_name)
+    except:
+        log.error('Timeout Error in finding Object in Sesame')
+        object_radec = SkyCoord(tel_pnt_sky_pos.icrs)
+
+    # Observation modes
+    source_pointing_diff = object_radec.separation(
+                SkyCoord(tel_pnt_sky_pos.ra, tel_pnt_sky_pos.dec)
+                ).deg
+    # Assuming wobble offset is fixed to 0.4
+    if round(source_pointing_diff, 1) == 0.4:
+        mode = 'WOBBLE'
+    elif round(source_pointing_diff, 1) > 1:
+        mode = 'OFF'
+    elif round(source_pointing_diff, 1) == 0.:
+        mode = 'ON'
+    else:
+        mode = 'ON-MISPOINTING' # Either this or modify the method of getting ON mode
+
```

Comment:
Any suggestions for naming a 'mispointing' mode of observation? There is no set standard right now for this in GADF, but it will be useful later for science tools, and decide upon the 'quality' of the data.

Reply:
UNDETERMINED? Or UNDERTERMINED-POINTING?

Reply:
I have kept it UNDETERMINED for now.